docs/src/writing-constructs.rst

```diff
@@ -0,0 +1,208 @@
+  query/modify the incoming resource as well. The latter is particularly
+  useful if something about IAM permissions need to be set, for example.
+* Use **XxxRef** instead of **Xxx** to allow imports to work; when accepting
+  resource parameters, accept the **XxxRef** variant of the class (if
+  available). This makes it possible for users to pass resources that aren't
+  defined in the same CDK app.
+* If your construct wraps a single (or most prominent) other construct, give it
+  an id of either **"Resource"** or **"Default"**; The "main" resource that an
+  AWS Construct represents should use the ID **Resource**, for higher-level
+  wrapping resources you will generally use **Default** (resources named
+  **Default** will inherit their parent's logical ID, while resources named
+  **Resource** will have a distinct logical ID but the human-readable part of it
+  will not show the "Resource" part).
+
+
+Code Organization
+=================
+
+Your package will roughly look like this:
+
+```
+your-package
+├── package.json
+├── README.md
+├── lib
+│   ├── index.ts
+│   ├── some-resource.ts
+│   └── some-other-resource.ts
+└── test
+    ├── integ.everything.lit.ts
+    ├── test.some-resource.ts
+    └── test.some-other-resource.ts
+
+```
+
+* Your package is named ``aws-xxx`` if it represents the canonical AWS
+  Construct Library for this service; otherwise we recommend starting with
+  ``cdk-``, but you are free to pick a pleasing name.
+* Code goes under **lib/**, tests go under **test/**.
+* Entry point should be **lib/index.ts** and should only contain ``export`` s
+  for other files.
+* No need to put every class in a separate file. Try to think of a
+  reader-friendly organization of your source files.
+* If you want to make package-private utility functions, put them in a file
+  that is *not exported* from **index.ts** and use that file as normal.
+* Free-floating functions (functions that are not part of a class definition)
+  cannot be accessed through **jsii** (i.e., from languages other than
+  TypeScript and JavaScript). Don't use them for public features of your
+  construct library.
+* Document all public APIs with doc comments (JSdoc syntax). Document defaults
+  using the **@default** marker in doc comments.
+
+Testing
+=======
+
+* Add unit tests for every construct (**test.xxx.ts**), relating the construct's
+  properties to the CloudFormation that gets generated. Use the
+  **@aws-cdk/assert** library to make it easier to write assertions on the
+  CloudFormation output.
+* Try to test one concern per unit test. Even if you *could* test more than one
+  feature of the construct per test, it's better to write multiple tests,
+  one for each feature. A test should have one reason to break.
+* Add integration tests (**integ.xxx.ts**) that are basically just CDK apps
+  which exercise the features of the construct, then load your shell with
+  credentials and run ``npm run integ`` to exercise them. You will also have to
```

Comment:
Why double quotes?

Reply:
Because that's monospace syntax: http://docutils.sourceforge.net/docs/user/rst/quickref.html

This is not MarkDown, it's ReST.